### PR TITLE
User/kmadisa/lmc 274/write unit tests for lmc dish sim

### PR DESCRIPTION
**Pull request by kmadisa**

### CODE REVIEW

File: tango_simlib/tests/test_dish.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        set_mode = dish_mode_enum_labels.index('OPERATE')
+        dish_mode_quant.last_val = set_mode
+        mock_time = Mock(return_value=float(timestamp))
+        self.model.time_func = mock_time
+        mock_set_val = Mock(
+            side_effect=dish_mode_quant.set_val)
+        dish_mode_quant.set_val = mock_set_val
+        calls = [call(dish_mode_enum_labels.index('CONFIG'), float(timestamp)),
+                 call(dish_mode_enum_labels.index('OPERATE'), float(timestamp))]
+        for cmd_name in ['ConfigureBand1', 'ConfigureBand2', 'ConfigureBand3',
```
Comment (brickZA) [inline tango_simlib/tests/test_dish.py]: - [ ] Perhaps use a different timestamp for each command.

File: tango_simlib/tests/test_dish.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+        # Write a value to the quantity to override the default one which is a boolean
+        # True value. Pick any value except for 'LOW'.
+        power_state_quant.last_val = power_state_enum_labels.index('OFF')
+
+        dish_mode_quant = self.model.sim_quantities['dishMode']
+        dish_mode_enum_labels = dish_mode_quant.meta['enum_labels']
+
+        for allowed_mode in ['STOW', 'MAINTENANCE']:
+            dish_mode_quant.last_val = dish_mode_enum_labels.index(allowed_mode)
```
Comment (brickZA) [inline tango_simlib/tests/test_dish.py]: - [ ] Reset the mode back to not-low in each iteration of the loop

File: tango_simlib/tests/test_dish.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    def test_set_maintenance_mode(self):
+        dish_mode_quant = self.model.sim_quantities['dishMode']
+        dish_mode_enum_labels = dish_mode_quant.meta['enum_labels']
+
+        # Write a value to the quantity to override the default one which is a boolean
+        # True value. Pick any value except for 'MAINTENACE'.
+        dish_mode_quant.last_val = dish_mode_enum_labels.index('STANDBY-LP')
+
+        for allowed_mode in ['STANDBY-LP', 'STANDBY-FP']:
+            dish_mode_quant.last_val = dish_mode_enum_labels.index(allowed_mode)
```
Comment (brickZA) [inline tango_simlib/tests/test_dish.py]: - [ ] Reset mode in each iteration.

File: tango_simlib/tests/test_dish.py [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+    def test_set_operate_mode(self):
+        dish_mode_quant = self.model.sim_quantities['dishMode']
+        dish_mode_enum_labels = dish_mode_quant.meta['enum_labels']
+
+        # Write a value to the quantity to override the default one which is a boolean
+        # True value. Pick any value except for 'MAINTENACE'.
+        dish_mode_quant.last_val = dish_mode_enum_labels.index('STANDBY-LP')
+
+        for allowed_mode in ['STANDBY-FP']:
+            dish_mode_quant.last_val = dish_mode_enum_labels.index(allowed_mode)
```
Comment (brickZA) [inline tango_simlib/tests/test_dish.py]: - [ ] Either don't use for, or make the loop correct for future use by resetting at the start.

File: tango_simlib/tests/test_dish.py [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
```
Comment (brickZA) [inline tango_simlib/tests/test_dish.py]: - [ ]  Same same about loop :)

File: tango_simlib/tests/test_dish.py [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+    def test_set_stow_mode(self):
+        dish_mode_quant = self.model.sim_quantities['dishMode']
+        dish_mode_enum_labels = dish_mode_quant.meta['enum_labels']
+        pointing_state_quant = self.model.sim_quantities['pointingState']
+        pointing_state_enum_labels = pointing_state_quant.meta['enum_labels']
+
+        self.model.sim_actions['SetStowMode']()
+        self.assertEqual(dish_mode_quant.last_val, dish_mode_enum_labels.index('STOW'))
+        self.assertEqual(pointing_state_quant.last_val,
+                         pointing_state_enum_labels.index('STOW'))
```
Comment (brickZA) [inline tango_simlib/tests/test_dish.py]: - [ ] Add a fake-time update loop to check that position quantities change as expected.

File: tango_simlib/tests/test_dish.py (hunk 5 above)
Comment (kmadisa) [inline tango_simlib/tests/test_dish.py]: Done.

File: tango_simlib/tests/test_dish.py (hunk 4 above)
Comment (kmadisa) [inline tango_simlib/tests/test_dish.py]: Done.

File: tango_simlib/tests/test_dish.py (hunk 2 above)
Comment (kmadisa) [inline tango_simlib/tests/test_dish.py]: Done.

File: tango_simlib/tests/test_dish.py (hunk 3 above)
Comment (kmadisa) [inline tango_simlib/tests/test_dish.py]: Done.

File: tango_simlib/tests/test_dish.py (hunk 1 above)
Comment (kmadisa) [inline tango_simlib/tests/test_dish.py]: Done.

File: tango_simlib/tests/test_dish.py [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+        cls.data_descr_files.append(
+            pkg_resources.resource_filename('tango_simlib.tests',
+                                            'DishElementMaster.xmi'))
+        cls.data_descr_files.append(pkg_resources.resource_filename(
+            'tango_simlib.tests', 'DishElementMaster_SIMDD.json'))
+        cls.device_name = 'test/nodb/tangodeviceserver'
+        cls.model = tango_sim_generator.configure_device_model(
+            cls.data_descr_files, cls.device_name)
+
+    def setUp(self):
```
Comment (kmadisa) [inline tango_simlib/tests/test_dish.py]: Done.

File: tango_simlib/tests/test_dish.py (hunk 6 above)
Comment (brickZA) [inline tango_simlib/tests/test_dish.py]: You've now added a list of mock times as a side effect in `test_set_stow_mode()` (GOOD!), but I don't see that you actually check that the model stored the correct time?

**Pull request merged**